portfolio/src/main/webapp/comments.js

```diff
@@ -13,24 +13,120 @@
-function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
+/**
+ * Fetches comments from the server and adds them to the DOM.
+ */
+function getComments() {
+  fetch(`/list-comments?${GET_PARAMS.toString()}`).then((response) => {
+    if (response.status === 200) {
+      response.json().then((comments) => {
+        const commentsContainer = document.querySelector('#comments-container');
+        
+        removeNodeChildren(commentsContainer);
+        addCommentsToDom(comments, commentsContainer);
+      });
+    }
+    else {
+      alert('Sorry, we could not process your request. ' +
+          'Reload the page and try again.');
+    }
+  });
+}
+
+/**
+ * Remove children of a HTML node
+ * @param {Object} node The HTML node, element.
+ */
+function removeNodeChildren(node) {
+  while (node.firstChild) {
+    node.removeChild(node.lastChild);
+  }
 }
 
-document.body.onload = getComments;
\ No newline at end of file
+/**
+ * Adds comments to a container using a template
+ * @param {!Array<{Object}>} comments The comments to add
+ * @param {Object} container The container to place comments inside
+ */
+function addCommentsToDom(comments, container) {
+  const commentTemplate = document.querySelector('#comment-card');
+
+  for (const comment of comments) {
+    const newComment = commentTemplate.content.cloneNode(true);
+
+    newComment.querySelector('h4').textContent = comment.username;
+    newComment.querySelector('span').textContent = comment.date;
+    newComment.querySelector('p').textContent = comment.comment;
+
+    container.appendChild(newComment);
+  }
+}
+
+/**
+ * Set up functions for when comment section events are triggered
+ */
+function setUpComments() {
+  // Listen to changes in the quantity of comments to display
+  document.getElementById('quantity').addEventListener('change', (event) => {
+    GET_PARAMS.set('quantity', event.target.value);
+    getComments();
+  });
+
+  // Listen to changes in the order to display comments
+  document.getElementById('dateOrder').addEventListener('change', (event) => {
+    GET_PARAMS.set('dateOrder', event.target.checked);
+    getComments();
+  });
+
+  // Listen to when the delete buttons is clicked
+  document.getElementById('delete-comments').addEventListener('click', (event) => {
+    deleteAllComments(event.target.value);
+  });
+}
+
+// Set up comments when DOM is loaded
+if (document.readyState === 'loading') {
+  // loading yet, wait for the event
+  document.addEventListener('DOMContentLoaded', () => {
+    setUpComments();
+    // Function in gallery.js
+    hasQueryString('imageId', () => {
+      GET_PARAMS.set('imageId', QUERY_STRING.get('imageId'))
+      getComments();
+    });
+  });
+} else {
+  // DOM is ready!
+  setUpComments();
+  // Function in gallery.js
+  hasQueryString('imageId', () => {
+    GET_PARAMS.set('imageId', QUERY_STRING.get('imageId'));
+    getComments();
+  });
+}
```

Comment:
Instead of listening for the 'loading' state, you can do something like below:

document.addEventListener('load', () => {
   setUpComments();
   hasQueryString();
});

That way the code only runs once the dom content has fully loaded.

Reply:
Okay, I am doing it this way, because 'DOMContentLoaded' is triggered when the document has been loaded and parsed (which can happen before adding the eventListener, that's why the 'loading' state is checked), without taking into account other resources, for example images. So you can start to apply javascript without waiting for images, etc. Meanwhile, 'load' is triggered when the whole page has loaded.

I guess that for our portfolio the different of time between these events is minimal, so there would be no problem to just use 'load', but I would like to hear your comments about it, Aki. Thanks!

Reply:
Just to reiterate here:

window.addEventListener('load', () => { ... }); if you need to wait for all resources to load

document.addEventListener('load', () => { ... }); if you only need to wait for the Dom content to load.

Reply:
Aki, I tried using `document.addEventListener('load', () => { ... });`, but it never triggered in my dev server. I tried moving the scripts to the `<head>` of the document, but even there it didn't trigger.

So I stick with `document.addEventListener('DOMContentLoaded', () => { ... });` , remove the listening for the `loading` state and I keep the scripts in the `<head>`, to make sure the scripts are loaded before continuing to parse the doc.